core/arch/arm/kernel/boot.c

```diff
@@ -1448,6 +1492,15 @@ void __weak boot_save_args(unsigned long a0, unsigned long a1 __unused,
 {
 	/*
 	 * Register use:
+	 *
+	 * Scenario A: FW Handoff via Transfer List
+	 * a0(x0/r2)	- DTB address or 0
+	 * a1(x1/r1)	- TRANSFER_LIST_SIGNATURE | REG_CONVENTION_VER_MASK
+	 * a2(x2/r0)	- must be 0
+	 * a3(x3/r3)	- Transfer list base address
+	 * a4(x4/r4)	- Not used
+	 *
+	 * Scenario B: Legacy Handoff arguments
```

Comment:
These are not legacy arguments. I'm not sure that Transfer List will be used with FF-A so it may give the wrong impression to list Transfer Lists as the primary scenario.

When `CFG_TRANSFER_LIST` is defined, it undeniably has precedence, but I don't think that will be defined for `CFG_CORE_SEL2_SPMC=y || CFG_CORE_EL3_SPMC=y`.

Reply:
I think we will use TL with FF-A, please refer to the discussion in the link: https://github.com/FirmwareHandoff/firmware_handoff/pull/19
We plan to add FF-A TEs e.g. `XFERLIST_SPMC_MANIFEST`.

But at the moment we don't support it yet.
Shall we still keep the term `Legacy Handoff arguments` here but add more comments to state that currently FF-A is not yet supported? Because I guess finally all 'standalone' arguments should become 'legacy'.

Reply:
Let's call these default arguments since that's what they are when TL isn't enabled or used.

Reply:
Sounds good.